raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java

```diff
@@ -108,63 +109,36 @@ public class QuorumConfig {
             .define(QUORUM_LINGER_MS_CONFIG, INT, DEFAULT_QUORUM_LINGER_MS, null, MEDIUM, QUORUM_LINGER_MS_DOC)
             .define(QUORUM_REQUEST_TIMEOUT_MS_CONFIG, INT, DEFAULT_QUORUM_REQUEST_TIMEOUT_MS, null, MEDIUM, QUORUM_REQUEST_TIMEOUT_MS_DOC)
             .define(QUORUM_RETRY_BACKOFF_MS_CONFIG, INT, DEFAULT_QUORUM_RETRY_BACKOFF_MS, null, LOW, QUORUM_RETRY_BACKOFF_MS_DOC);
-
-    private final int requestTimeoutMs;
-    private final int retryBackoffMs;
-    private final int electionTimeoutMs;
-    private final int electionBackoffMaxMs;
-    private final int fetchTimeoutMs;
-    private final int appendLingerMs;
-
-    public QuorumConfig(AbstractConfig abstractConfig) {
-        this(
-            abstractConfig.getInt(QUORUM_REQUEST_TIMEOUT_MS_CONFIG),
-            abstractConfig.getInt(QUORUM_RETRY_BACKOFF_MS_CONFIG),
-            abstractConfig.getInt(QUORUM_ELECTION_TIMEOUT_MS_CONFIG),
-            abstractConfig.getInt(QUORUM_ELECTION_BACKOFF_MAX_MS_CONFIG),
-            abstractConfig.getInt(QUORUM_FETCH_TIMEOUT_MS_CONFIG),
-            abstractConfig.getInt(QUORUM_LINGER_MS_CONFIG)
-        );
     }
 
     public QuorumConfig(
-        int requestTimeoutMs,
-        int retryBackoffMs,
-        int electionTimeoutMs,
-        int electionBackoffMaxMs,
-        int fetchTimeoutMs,
-        int appendLingerMs
+            Map<?, ?> props
     ) {
-        this.requestTimeoutMs = requestTimeoutMs;
-        this.retryBackoffMs = retryBackoffMs;
-        this.electionTimeoutMs = electionTimeoutMs;
-        this.electionBackoffMaxMs = electionBackoffMaxMs;
-        this.fetchTimeoutMs = fetchTimeoutMs;
-        this.appendLingerMs = appendLingerMs;
+        super(configDef(), props);
     }
 
     public int requestTimeoutMs() {
-        return requestTimeoutMs;
+        return getInt(QUORUM_REQUEST_TIMEOUT_MS_CONFIG);
```

Comment:
I'm re-thinking the design about inheritance and composition.  `_remoteLogManagerConfig` has a issue that it can't reflect the updated configs (#16394) and I left a comment (https://github.com/apache/kafka/pull/16394#discussion_r1646273869) about keeping a `KafkaConfig` reference instead of copying all props of `KafkaConfig`. 

It seems all configs in `QuorumConfig` can't be updated dynamically, so it is fine for now. However, I'd like to see more consistent design for those individual config class. Hence, maybe we can take a look at #16394 first, and then this PR can follow the same pattern :)

Reply:
thanks @OmniaGM @chia7712 
let me check https://github.com/apache/kafka/pull/16394 first

Reply:
just check https://github.com/apache/kafka/pull/16394 and it looks like currently the approach is instead of inheriting, we composite an AbstractConfig as a property of the actual config. 
I am wondering if in the future we can put all config related implementation in a module, instead of scattering around in different modules. In this case, take this QuorumConfig for example, we can have a constructor that accept KafkaConfig to init the QuorumConfig.

Reply:
I believe this is the discussion @chia7712 and I are having in https://github.com/apache/kafka/pull/16394 specially that we are trying to pull KafkaConfig out of core.